neurodsp/sim/combined.py

```diff
@@ -82,3 +84,89 @@ def sim_combined(n_seconds, fs, components, component_variances=1):
     sig = np.sum(sig_components, axis=0)
 
     return sig
+
+
+@normalize
+def sim_central_freq(n_seconds, fs, central_freq, bw, ht, ap_func='sim_powerlaw',
+                     ap_kwargs={'exponent': -2}):
+    """
+    Returns a time series whose full power spectrum consists of an aperiodic component
+    and a gaussian peak at central_freq with standard deviation bw and relative height ht.
+
+
+    Parameters
+    -----------
+    n_seconds: float
+        Number of seconds elapsed in the time series.
+    fs: float
+        Sampling rate.
+    central_freq: float
+        Central frequency for the gaussian peak in Hz.
+    bw: float
+        Bandwidth, or standard deviation, of gaussian peak in Hz.
+    ht: float
+        Relative height in log_10(Hz) over the aperiodic component of the gaussian peak at
+        central_freq.
+    ap_func : str, optional, default: 'sim_powerlaw'
+        An aperiodic simulation function from ``neurodsp.sim.aperiodic``.
+    ap_kwargs : dict, optional, default: {'exponent': -2}
+        Keyword arguments to use with ``ap_func``.
+
+    Returns
+    -------
+    sig: 1d array
+        Time series with desired power spectrum.
+
+    Examples
+    --------
+    Simulate an aperiodic component with exponent -2 superimposed over
+    an oscillatory component with central frequency 20.
+
+    >>> sig = sim_central_freq(n_seconds=50, fs=500, central_freq=20, bw=5, ht=7,
+    ...                        ap_func='sim_powerlaw', ap_params={'exponent': -2})
+    """
+
+    times = create_times(n_seconds, fs)
+
+    # Construct the aperiodic component and compute its Fourier transform. Only use the
+    # first half of the frequencies from the Fourier transform since the signal is real.
+    sig_components = {ap_func: ap_kwargs}
+    sig_ap = sim_combined(n_seconds, fs, sig_components)
+    sig_len = sig_ap.shape[0]
+
+    sig_ap_hat = np.fft.fft(sig_ap)[0:(sig_len//2+1)]
+
+    # Create the range of frequencies that appear in the power spectrum since these
+    # will be the frequencies in the cosines we sum below
+    freqs = np.linspace(0, fs/2, num=sig_len//2 + 1, endpoint=True)
+
+    # Construct the array of relative heights above the aperiodic power spectrum
+    rel_heights = np.array([ ht * np.exp(-(f - central_freq)**2/(2*bw**2)) for f in freqs])
+
+    # Build an array of the sum of squares of the cosines as they appear in the calculation of the
+    # amplitudes
+    cosine_norms = np.array([
+                          norm(
+                            np.cos(2*np.pi*f*times), 2
+                          )**2 for f in freqs
+                        ])
+
+    # Build an array of the amplitude coefficients
+    cosine_coeffs = np.array([
+                    (-np.real(sig_ap_hat[ell]) + np.sqrt(np.real(sig_ap_hat[ell])**2 + \
+                        (10**rel_heights[ell] - 1)*np.abs(sig_ap_hat[ell])**2))/cosine_norms[ell]
+                    for ell in range(cosine_norms.shape[0])]
+                )
```

Comment:
holy math :)

I'm not going to pretend I really understand what's going on here. This might be a target for a bit more inline comment description?

also, style comment: multi-line spacing is a bit off (I think - easier to check in text-editor). One of us can run a lint and style check at some point.